### PR TITLE
Fixes #31277 - optimize css files

### PR DESCRIPTION
**Pull request by amirfefer**
Following  https://github.com/theforeman/foreman/pull/8123, I separated the webpack optimization to here
Before:
before

After:
After

### CODE REVIEW

File: config/webpack.config.js [hunk 1]
```diff
@@ -3,41 +3,44 @@
 
 var path = require('path');
 var webpack = require('webpack');
-var ForemanVendorPlugin = require('@theforeman/vendor').WebpackForemanVendorPlugin;
+var ForemanVendorPlugin = require('@theforeman/vendor')
```
Comment (ekohl) [inline config/webpack.config.js:6]: Formatting changes in the same commit as actual changes is very painful for a reviewer. Please split it out into two commits.

**Review by ekohl — CHANGES_REQUESTED** [review]
Please open a packaging PR to add `optimize-css-assets-webpack-plugin`.

File: package.json [hunk 2]
```diff
@@ -47,6 +47,7 @@
     "file-loader": "^0.9.0",
     "highlight.js": "~9.14.0",
     "node-sass": "^4.5.0",
+    "optimize-css-assets-webpack-plugin": "^3.2.0",
```
Comment (amirfefer) [inline package.json:50]: Created a packaging PR - https://github.com/theforeman/foreman-packaging/pull/5980

**Comment by ezr-ondrej** [comment]
In general, I quite like this change :+1: Altough I believe that we compile unecessary stuff in the global files, the import from files are circularly importing almost the whole forman js/css, we should take a look at these dependencies and make them as sane as possible.
So at least the imports of `register<MyGlobalService/Route/Component/>` methods won't cause these to grow. I've noticed, that for my fill.css had all the patternfly styling, I doubt it is neccessary.
This change is still relevant tho :+1:
I'd like to see it for production only, as in develop it is better to see what gets included.

File: package.json (hunk 2 above)
Comment (amirfefer) [reply package.json:50]: Merged

**Comment by amirfefer** [comment]
@ekohl  thanks for the review
this package has been merged -  theforeman/foreman-packaging#5980
could you please re-review?

File: package.json (hunk 2 above)
Comment (ezr-ondrej) [reply package.json:50]: Why we haven't included it in the vendor/builder only? :thinking:

**Review by ezr-ondrej — DISMISSED** [review]
I'm ok with this and it works in development.
But I'd like this to work only in production and I don't know how to test that.
@theforeman/packaging is it enough `rake webpack:compile` worked fine? :)

**Review by ezr-ondrej — DISMISSED** [review]

**Comment by ezr-ondrej** [comment]
@ekohl could you take a look or would you know who would?

**Review by ekohl — DISMISSED** [review]
Looks like the nightly pipelines are green now so we can see if this actually makes a difference.

**Comment by ezr-ondrej** [comment]
Ah... there was missing space in one commit message :facepalm:
I've fixed that. @ekohl could you please ack again?

**Review by ekohl — DISMISSED** [review]
In this case I prefer multiple commits so the failed check for a single commit should be ignored. It should also **not** be squashed on merge IMHO.

**Comment by Ron-Lavi** [comment]
can someone rebase?
hope to get this in soon :)

**Comment by ezr-ondrej** [comment]
@ekohl I've missed the window, but I guess better we waited for @amirfefer to be back in case something goes wrong. Could you please reack? :pleading_face:

**Review by ekohl — DISMISSED** [review]
I'd wait with merging until nightly is green again, but in itself it looks ok.

**Comment by ezr-ondrej** [comment]
While waiting there has been one more conflict introduced :(

**Comment by ezr-ondrej** [comment]
[test upgrade]

**Review by ezr-ondrej — APPROVED** [review]

**Review by tbrisker — APPROVED** [review]
Thanks @amirfefer !

**Comment by tbrisker** [comment]
Please update the foreman package dependency to pull in the new package

**Pull request merged**